src/components/Geogebra/StudentClient.js

```diff
@@ -0,0 +1,453 @@
+            width: 800,
+            height: 600,
+            perspective: 'AG',
+            showAlgebraInput: true,
+            showToolBarHelp: false,
+            showMenubar: true,
+            enableLabelDrags: false,
+            showResetIcon: true,
+            showToolbar: true,
+            allowStyleBar: false,
+            useBrowserForJS: true,
+            enableShiftDragZoom: true,
+            errorDialogsActive: true,
+            enableRightClick: false,
+            enableCAS: false,
+            enable3d: false,
+            isPreloader: false,
+            screenshotGenerator: false,
+            preventFocus: false,
+            scaleContainerClass: 'appletContainer',
+            resizeFactor: 1.0,
+            // Initial Geogebra state encoded in Base64
+            initialState,
+            ...params,
+        };
+
+        this.log = this.params.log;
+        this.params.listener = undefined;
+        this.appletContainer = new GGBApplet(this.params);
+        this.ignoreUpdates = false;
+    }
+
+    /**
+     * Called by the client to start initializing the applet.
+     *
+     * @param listener Observer object, notified of important events such as
+     * `onAddElement`.
+     */
+    async initApplet(listener) {
+        this.log.debug(listener);
+
+        this.listener = listener;
+
+        // Geogebra uses global callbacks.
+        window.ggbOnInit = this.ggbOnInit.bind(this);
+
+        // Start applet render. When finished, `window.ggbOnInit` will be
+        // called.
+        this.appletContainer.inject(this.params.container, 'auto');
+    }
+
+    /**
+     * Internal callback for successful initialization of the applet.
+     * Finishes initialization process and notifies listener.
+     */
+    ggbOnInit() {
+        this.log.debug();
+
+        // Save reference to the aplet itself. This is the main line
+        // of communication between Geogebra and the rest of the code.
+        this.applet = this.appletContainer.getAppletObject();
+
+        // Geogebra uses global listners on `window` to nofity of its
+        // internal events (such as when element is added/modified/deleted).
+        this.registerGlobalListeners();
+
+        // Resize applet on window resize.
+        window.onresize(() => {
+            this.applet.setHeight(window.innerHeight / this.params.resizeFactor);
+        });
+
+        // Notify listener.
+        this.listener.onAppletReady();
+    }
+
+    /**
+     * Geogebra uses global listners on `window` to nofity of its
+     * internal events (such as when element is added/modified/deleted).
+     * This methods sets and links those listeners with this object.
+     *
+     * Call chain is as follows:
+     * Geogebra -> Window -> Client
+     */
+    registerGlobalListeners() {
+        // Setup `window` methods which refer to this object.
+        window[`addListener${this.appletId}`] = throttle(
+            label => this.onAddElement(label),
+            50
```

Comment:
Is it possible to have the throttle value specified as a const somewhere? Hardcoded 50 is iffy